cmd/cloud/utils.go

```diff
@@ -0,0 +1,143 @@
+package cloud
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/safedep/vet/internal/auth"
+	"github.com/safedep/vet/internal/ui"
+	"gopkg.in/yaml.v2"
+)
+
+type TokenResponse struct {
+	AccessToken string `json:"access_token"`
+	ExpireIn    int    `json:"expires_in"`
+	Scope       string `json:"scope"`
+	IDToken     string `json:"id_token"`
+	Tokentype   string `json:"token_type"`
+}
+
+func loadConfiguration() (auth.Config, error) {
+	path, err := os.UserHomeDir()
+	if err != nil {
+		return auth.Config{}, err
+	}
+
+	path = filepath.Join(path, auth.HomeRelativeConfigPath())
+
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return auth.Config{}, err
+	}
+
+	var config auth.Config
+	err = yaml.Unmarshal(data, &config)
+	if err != nil {
+		return auth.Config{}, fmt.Errorf("config deserialization failed: %w", err)
+	}
+
+	return config, nil
+}
+
+func persistConfiguration(config auth.Config) error {
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("config serialization failed: %w", err)
+	}
+
+	path, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+
+	path = filepath.Join(path, auth.HomeRelativeConfigPath())
+
+	err = os.MkdirAll(filepath.Dir(path), os.ModePerm)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(path, data, 0o600)
+}
+
+func getNewAccessTokenAndPersistIt(config auth.Config) error {
+	data := url.Values{}
+	data.Set("grant_type", "refresh_token")
+	data.Set("refresh_token", config.CloudRefreshToken)
+	data.Set("client_id", auth.CloudIdentityServiceClientId())
+
+	req, _ := http.NewRequest("POST", auth.CloudIdentityServiceTokenUrl(), strings.NewReader(data.Encode()))
```

Comment:
Try using the `cli/oauth` package to achieve this instead of making manual HTTP call. See `cmd/cloud/quickstart.go`

Reply:
I check the `cli/oauth` package and can see that it is either used to `deviceCode` or `webflow` as in our case we don't want user to redirect anywhere to get the new access token. 

So I don't think we can use `cli/oauth` package  here.